### PR TITLE
txn: return the action check_txn_status does

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
Return the action `check_txn_status` does. `check_txn_status` should tell the caller whether the `min_commit_ts` is pushed or not. Only when it's pushed, the caller can bypass the lock.

###  What is the type of the changes?
- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

###  Refer to a related PR or issue link (optional)
https://github.com/pingcap/tidb/pull/13432

### CODE REVIEW

**Comment by youjiali1995** [comment]
@coocood @tiancaiamao PTAL

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -2228,13 +2228,19 @@ fn future_check_txn_status<E: Engine, L: LockMgr>(
...
+                    TxnStatus::TtlExpire => resp.set_action(Action::TtlExpireRollback),
+                    TxnStatus::LockNotExist => resp.set_action(Action::LockNotExistRollback),
                     TxnStatus::Committed { commit_ts } => resp.set_commit_version(commit_ts),
-                    TxnStatus::Uncommitted { lock_ttl } => resp.set_lock_ttl(lock_ttl),
+                    TxnStatus::Uncommitted {
+                        lock_ttl,
+                        min_commit_ts,
+                    } => {
+                        resp.set_lock_ttl(lock_ttl);
+                        if min_commit_ts != 0 {
```
Comment (MyonKeminta) [inline src/server/service/kv.rs]: How about `min_commit_ts > req.get_caller_start_ts()` ?

Comment (coocood) [reply src/server/service/kv.rs]: But we need to know if it is safe to non-blocking read.

Comment (MyonKeminta) [reply src/server/service/kv.rs]: Isn't this safe?

Comment (youjiali1995) [reply src/server/service/kv.rs]: As long as the `min_commit_ts > caller_start_ts`, it's safe.

**Comment by coocood** [comment]
LGTM

**Pull request merged**